mesonbuild/compilers/c.py

```diff
@@ -147,6 +151,7 @@ class AppleClangCCompiler(ClangCCompiler):
 
     _C17_VERSION = '>=10.0.0'
     _C18_VERSION = '>=11.0.0'
+    _C2X_VERSION = '>=12.0.0'
```

Comment:
Note that this is guesswork. I don't have a Mac to check, so I have no actual idea if Xcode 12 ships with the Apple equivalent of Clang 9.